go/plugins/compat_oai/compat_oai.go

```diff
@@ -0,0 +1,154 @@
+package compat_oai
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/firebase/genkit/go/genkit"
+	openaiGo "github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+)
+
+var (
+	// BasicText describes model capabilities for text-only GPT models.
+	BasicText = ai.ModelInfoSupports{
+		Multiturn:  true,
+		Tools:      true,
+		SystemRole: true,
+		Media:      false,
+	}
+
+	// Multimodal describes model capabilities for multimodal GPT models.
+	Multimodal = ai.ModelInfoSupports{
+		Multiturn:  true,
+		Tools:      true,
+		SystemRole: true,
+		Media:      true,
+	}
+)
+
+// State management
+var state struct {
+	mu      sync.Mutex
+	initted bool
+	client  *openaiGo.Client
+}
+
+func OpenAICompatible(ctx context.Context, g *genkit.Genkit, provider string, opts ...option.RequestOption) error {
```

Comment:
Do you think we should add some sort of typing/checking for which providers are supported here?

Reply:
I think we should allow all providers, it is up to the users to know which ones they wanna use so we do not limit them in that way